### ISSUE TITLE
Add setting to turn off intelligent spacebar

### ISSUE THREAD

**Issue by saintsoldierx** [KhalisFoundation/sttm-desktop#1761]
When off, it should always just go to home pankti

**Comment by Gauravjeetsingh** [comment]
@saintsoldierx veerji, we can definetly do that, but I am a little hesistant in adding a setting for this. If the spacebar is not being used intelligently, maybe have two separate shortcuts? One for intelligent scrolling and keep space bar for home pankti?

It could be **shift + space**, its intelligent scrolling
or any other single key event, like **i**

**Comment by saintsoldierx** [comment]
No, i disagree - reason being is that spacebar is the most easiest key on the keyboard to use and our users are not only used to it, but it seems to be more intuitive than any other key. So i would say to add a setting for this and give users the option to switch.

**Comment by singh-sp** [comment]
Waheguru ji,

I have the change ready that allows users to disable the (not so) "intelligent spacebar" feature. Note: I have left it enabled by default.

I am sure it has caused many others such a hassle since it was released. 😅- encountered numerous bugs (I was too lazy to report them).

Preview:

https://github.com/KhalisFoundation/sttm-desktop/assets/6597802/fac72d13-9885-4a27-9616-bd39e39330e6

**Comment by saintsoldierx** [comment]
@singh-sp thank you, can you create a PR? I am wondering though why you don't like the intelligent spacebar feature? It's designed for Keertan where you can use it auto switch from asthaees to antras and back again easily.

**Comment by singh-sp** [comment]
> @singh-sp thank you, can you create a PR? I am wondering though why you don't like the intelligent spacebar feature? It's designed for Keertan where you can use it auto switch from asthaees to antras and back again easily. 

Sure! I will create a PR tomorrow.

The reason we (myself and others) don't like it is because it forces you to follow a specific order. In reality, a Raagi can sing a shabad in *any order and may choose to revisit some panktis. When that happens, the spacebar becomes completely useless and out of sync, which is incredibly frustrating!

**Comment by saintsoldierx** [comment]
@singh-sp sure i get that. But surely it works in many situations too? Would it be beneficial then to have a shortcut as well to enable/disable it rather than going to settings each time?

**Comment by singh-sp** [comment]
Perhaps it works for some folks, and that’s great!

Personally, I don’t feel confident using it after noticing many mistakes it has made, such as going to Salok or Paudi Panktis. Since the entire sangat is looking at the projector screen and our raagis rely on the monitor in front of them, it is crucial that we don’t display the wrong panktis. With intelligent spacebar, knowing which pankti it will select next is impossible—it’s a hit or miss. While we could add a UI indicator to show the next pankti, it might clutter the UI. I love the way the app works without it—total control.

So overall, it’s a permanent preference and, therefore, an option in settings suits better.

**Comment by Gauravjeetsingh** [comment]
The fix is now merged in beta release, closing this out.
